LucidTestKit/Doubles/GraphStub.swift

```diff
@@ -60,6 +62,14 @@ public final class GraphStub: MutableGraph {
         return entityRelationshipSpies[identifier] != nil
     }
 
+    public func setEndpointResultMetadata(_ metadata: EndpointResultMetadata) {
+        self._metadata = metadata
+    }
+
+    public func metadata<E>() -> Metadata<E>? where E : Entity {
+        return self._metadata.map { Metadata<E>($0) }
+    }
```

Comment:
nitpick: we only use `self.` in `init` and in places where it denotes a retain (e.g. inside of a block).